Range Slider + Classics

gardaholm opened the pull request:
Adressed Issues #40 and first part for #39
Range filter is working, tested it a lot and made it dynamic, based on the difficulty_grades from da sqlite.

Pretty sure that the Javascript at the bottom can be optimized a bit. But I'm already pretty happy with the result. Would be great if you could have a look and test it – thanks! Did some testing and compared results with the app.

Classics are being filtered, but didn't start with displaying the icon on results.

lemeryfertitta commented on climbdex/static/css/nouislider.css:
```diff
@@ -0,0 +1,303 @@
+/* Functional styling;
```
This is really slick! I think its way better than the double select input - big improvement here. Looks pretty solid visually to me. There were two small things I noticed from a usability perspective that would be nice to fix if possible. I'm not super familiar with the plugin so I don't know how easy/doable these things are:

1. It would be great for the back button to still work for the range slider. I think resetting to min/max is a bit of a nuisance. This _should_ be possible if the initial values are based on an `input` item in the `form`, but it seems like the inputs are maybe getting set by JS so they are valueless when you go back.
2. When min=max (i.e. filtering by exact grade), you can only move the slider in the direction of the last side that you moved. So if you brought the min up to the max, you can only drop the min, not raise the max. It's not a huge deal but I feel like it means there is 50%-ish chance of trying to move the slider and failing in this scenario. It would be great if the slider was just flexible and let you drag in either direction when min=max. Not sure if this is something the nouislider supports, but would be cool if it did.

I think 1. should definitely be fixable. If 2. isn't fixable its not a big deal, I'd still push the changes without that and just have a separate issue to improve

gardaholm replied:
thanks, I also like it! really good useability addon.

> 1. It would be great for the back button to still work for the range slider. I think resetting to min/max is a bit of a nuisance. This _should_ be possible if the initial values are based on an `input` item in the `form`, but it seems like the inputs are maybe getting set by JS so they are valueless when you go back.

should be possible, I just tested it within the same window and this worked here, but probably chrome does save the initial values. if you share the link and go back things are reset. but we can prefill the slider positions based on the values from the url

> 2. When min=max (i.e. filtering by exact grade), you can only move the slider in the direction of the last side that you moved. So if you brought the min up to the max, you can only drop the min, not raise the max. 

maybe we can set the slider so if min=max then the two handles don't match but stay exactly next to each other. but then the labels just be combined to just one. maybe this is doable. should be a common case actually.

lemeryfertitta replied:
> I just tested it within the same window and this worked here

Huh that's strange, it does seem to be working now. When I tested yesterday it wasn't working, but cool if it does. No need to worry about behavior when coming from a link.

>  but stay exactly next to each other.

Just saw your commit here, nice improvement! It still has the same difficulty with moving the slider when they meet, but at least the label is really clear now. Personally, I'm happy with leaving it like this for now. I think it is strictly an improvement over the current UI.

lemeryfertitta reviewed (commented):
Just was testing the PR visually and had a comment on the slider, will take a look at the code later.

gardaholm commented:
Added Classic Icons for Issue #39 
used the © as you've mentioned, not the (b).
Updated the Advanced Filter to only be shown for Decoy + Tension Boards, I think these are the only two boards which support classics?

lemeryfertitta commented:
> Added Classic Icons for Issue #39 used the © as you've mentioned, not the (b). Updated the Advanced Filter to only be shown for Decoy + Tension Boards, I think these are the only two boards which support classics?

Nice! I'll double check on the classics. I also think it may just be worth showing the "hidden" Kilter classics for a consistent UI.

I'm away on a trip at the moment but I'll do a closer code review on this and try to get these features merged and deployed in the next couple weeks when I'm back.

lemeryfertitta reviewed (changes requested):
@gardaholm Sorry about dropping the ball on this one. Went on a trip and then had a bit less motivation to work on Climbdex after my return. I have access to the TB2 now and it has renewed my psych a bit 😄 .

Can we trim this PR down to just be the range slider? I'd like to handle the swipe changes in a different way, as I think introducing a library for gestures will help in the long run. It will also be a bit easier for me to review and add the classics feature right away if that was just in a separate, smaller PR. I'd like to play with the slider a bit more before releasing.

lemeryfertitta commented on climbdex/static/js/filterSelection.js:
```diff
@@ -118,3 +92,90 @@ if (document.referrer) {
+}
+
+function convertGrade(value) {
+  // Überprüfen, ob der Wert im Mapping existiert
+  if (gradeMapping[value]) {
+    // Wenn ja, geben Sie den konvertierten Wert zurück
+    return gradeMapping[value];
+  }
+
+  // Wenn der Wert nicht im Mapping existiert, Option zur Handhabung
+  console.warn('Grade conversion not found for:', value);
+  return null; // oder geben Sie den ursprünglichen Wert zurück, falls bevorzugt
```
I'd like to keep the repo in English 😄 - happy to have the translated versions of these comments or just have them removed if not necessary.

gardaholm commented:
> @gardaholm Sorry about dropping the ball on this one. Went on a trip and then had a bit less motivation to work on Climbdex after my return. I have access to the TB2 now and it has renewed my psych a bit 😄 .
> 
same here :)
and yes the TB2 is ace!

Will fix the issues and remove my german comments (forgot to remove them firsthand …)

gardaholm reviewed (commented):
Changes in touchsensitivity are reverted (swipes are like commited in #42)
#40 and #39 are still fixed with this pr

lemeryfertitta reviewed (approved):
Thanks! I made a few code styling and organization updates but otherwise LGTM. I tested a bit on my laptop and phone and feel like its working pretty well. I'll merge this, test it one more time, and then put out a release for it. Appreciate the hard work on this one!
